### PR TITLE
[ticket/15372] Add permission for listing topics without being able to read them

### PR DESCRIPTION
**Pull request by DavidIQ**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15372

### CODE REVIEW

File: phpBB/phpbb/db/migration/data/v32x/f_list_topics_permission_add.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+			array('permission.add', array('f_list_topics', false)),
+		);
+	}
+
+	public function revert_data()
+	{
+		return array(
+			array('permission.remove', array('f_list_topics', false)),
+		);
+	}
```
Comment (Elsensee) [inline phpBB/phpbb/db/migration/data/v32x/f_list_topics_permission_add.php]: The `update_data()` method will be automatically reverted and thus this method is unnecessary.

File: phpBB/language/en/acp/permissions_phpbb.php [hunk 2]
```diff
@@ -107,6 +107,7 @@
 // Forum Permissions
 $lang = array_merge($lang, array(
 	'ACL_F_LIST'		=> 'Can see forum',
+	'ACL_F_LIST_TOPICS' => 'Can see topics',
```
Comment (Elsensee) [inline phpBB/language/en/acp/permissions_phpbb.php]: I would find this really confusing.

- Can see forum
- Can see topics
- Can read forum

But you can read the forum if you can see the topics, right? Right? #AdvocatusDiaboli

Comment (DavidIQ) [reply phpBB/language/en/acp/permissions_phpbb.php]: Not really, no. If you can see topics but can't read the forum then you won't be able to open individual topics even though you'd see the titles.

**Comment by DavidIQ** [comment]
Totally agree on redoing the auth system in a future version. It's a very large undertaking however and will be very time consuming to even begin.

File: phpBB/phpbb/db/migration/data/v32x/f_list_topics_permission_add.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	{
+		return array(
+			'\phpbb\db\migration\data\v32x\v321',
+		);
+	}
+
+	public function update_data()
+	{
+		return array(
+			array('permission.add', array('f_list_topics', false)),
```
Comment (Elsensee) [inline phpBB/phpbb/db/migration/data/v32x/f_list_topics_permission_add.php]: Actually, can you copy the permission from `f_read`? I think that might make sense.

Comment (DavidIQ) [reply phpBB/phpbb/db/migration/data/v32x/f_list_topics_permission_add.php]: As in grant everyone with `f_read` the `f_list_topics` permission as well?

Comment (Elsensee) [reply phpBB/phpbb/db/migration/data/v32x/f_list_topics_permission_add.php]: Yep. Just so we have a standard value for existing installations that does make sense.

**Comment by marc1706** [comment]
@Elsensee I guess this looks good to you as well now?

**Pull request merged**